consul/consul.go

```diff
@@ -40,20 +40,25 @@ func (c *Consul) GetServices(name string) ([]*consulapi.CatalogService, error) {
 }
 
 func (c *Consul) getServicesUsingProviderWithRetriesOnAgentFailure(provide ServicesProvider) ([]*consulapi.CatalogService, error) {
-	var services []*consulapi.CatalogService = nil
-	for {
-		agent, agentAddress, err := c.agents.GetAnyAgent()
+	for retry := uint32(0); retry <= c.config.GetServicesRetries; retry++ {
+		agent, err := c.agents.GetAnyAgent()
 		if err != nil {
 			return nil, err
 		}
-		if services, err = provide(agent); err != nil {
-			log.WithError(err).WithField("Address", agentAddress).Error("An error occurred getting services from Consul, retrying with another agent")
-			c.agents.RemoveAgent(agentAddress)
+		if services, err := provide(agent.Client); err != nil {
+			log.WithError(err).WithField("Address", agent.IP).
+				Error("An error occurred getting services from Consul, retrying with another agent")
+			if failures := agent.IncFailures(); failures > c.config.MaxAgentGetServicesFailures {
```

Comment:
On the other hand, after renaming the config property to "failure tolerance" this makes sense, 0 means no tolerance.